### PR TITLE
Nour/proportional priority fees

### PR DESCRIPTION
**Pull request by NourAlharithi**

### CODE REVIEW

File: src/experimental-bots/filler/fillerMultithreaded.ts [hunk 1]
```diff
@@ -282,21 +285,29 @@ export class FillerMultithreaded {
 		this.blockhashSubscriber = new BlockhashSubscriber({
 			connection: driftClient.connection,
 		});
-		this.priorityFeeSubscriber = priorityFeeSubscriber;
+
+		const perpMarkets = this.marketIndexesFlattened.map((m) => {
```
Comment (crispheaney) [inline src/experimental-bots/filler/fillerMultithreaded.ts:289]: nit: technically not perpMarkets?

File: src/experimental-bots/filler-common/utils.ts [hunk 2]
```diff
@@ -370,6 +376,47 @@ export const getDriftClientFromArgs = ({
 	return driftClient;
 };
 
+export const getUserFeeTier = (
```
Comment (crispheaney) [inline src/experimental-bots/filler-common/utils.ts:379]: is this not in sdk?

File: src/experimental-bots/filler/fillerMultithreaded.ts [hunk 3]
```diff
@@ -1644,7 +1669,39 @@ export class FillerMultithreaded {
 			takerUserSlot,
 			referrerInfo,
 			marketType,
+			fillerRewardEstimate,
 		} = await this.getNodeFillInfo(nodeToFill);
+
+		if (!buildForBundle) {
+			const solPrice = this.driftClient.getOracleDataForPerpMarket(0).price;
+			const fillerRewardMicroLamports = Math.floor(
+				fillerRewardEstimate
```
Comment (crispheaney) [inline src/experimental-bots/filler/fillerMultithreaded.ts:1678]: this code is duped, should it be in it's own fn?

File: src/experimental-bots/filler/fillerMultithreaded.ts [hunk 4]
```diff
@@ -2018,13 +2082,31 @@ export class FillerMultithreaded {
...
+	 * @param quoteAssetAmount
+	 */
+	protected calculateFillerRewardEstimate(
+		feeTier: FeeTier,
+		quoteAssetAmount: BN
+	) {
+		const takerFee = quoteAssetAmount
+			.muln(feeTier.feeNumerator)
+			.divn(feeTier.feeDenominator);
+		const fillerReward = BN.min(new BN(10_000), takerFee.muln(10).divn(100));
```
Comment (crispheaney) [inline src/experimental-bots/filler/fillerMultithreaded.ts:2098]: nit: can just divide taker fee by 10?

File: src/experimental-bots/filler/fillerMultithreaded.ts [hunk 5]
```diff
@@ -2084,13 +2166,32 @@ export class FillerMultithreaded {
...
+				MarketType.PERP,
+				this.driftClient.getStateAccount(),
+				(
+					await this.userStatsMap.mustGet(takerUserAccount.authority.toString())
+				).getAccount()
+			),
+			// eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+			nodeToFill.node
+				.order!.price.mul(nodeToFill.node.order!.baseAssetAmount)
+				.mul(QUOTE_PRECISION)
```
Comment (crispheaney) [inline src/experimental-bots/filler/fillerMultithreaded.ts:2180]: are you intending to multply by price?

File: src/experimental-bots/filler/fillerMultithreaded.ts [hunk 6]
```diff
@@ -2084,13 +2166,32 @@ export class FillerMultithreaded {
...
+				.mul(QUOTE_PRECISION)
```
Comment (crispheaney) [inline src/experimental-bots/filler/fillerMultithreaded.ts:2181]: extra multiply

File: src/experimental-bots/filler/fillerMultithreaded.ts (hunk 4 above)
Comment (NourAlharithi) [reply src/experimental-bots/filler/fillerMultithreaded.ts:2098]: was being safe for precision's sake. but will change to divn

File: src/experimental-bots/filler-common/utils.ts (hunk 2 above)
Comment (NourAlharithi) [reply src/experimental-bots/filler-common/utils.ts:379]: it's on the sdk but on a User object which we dont already create -- so i just copied it out so we wouldnt have to make a User using the OneShotSubscriber and an extra subscribe just to access this method

File: src/experimental-bots/filler/fillerMultithreaded.ts (hunk 1 above)
Comment (NourAlharithi) [reply src/experimental-bots/filler/fillerMultithreaded.ts:289]: changed

**Pull request merged**